### ISSUE TITLE
False positive rate more if increase resolution

### ISSUE THREAD

**Issue by shuchitagupta** [AlexeyAB/Yolo_mark#25]
Hi,

As you suggested, to detect smaller objects I increased the resolution to 1024x1024. But this increases my false positive rate. Could you suggest how to reduce this? Also, how should training be done if my object size is small, that is street light. I am also cropping the images with street lights for training and resizing to 416x416 to use for training. Please suggest. Thanks.

**Comment by AlexeyAB** [comment]
Hi,

* To avoid false-positives: resolution for training and for detection shouldn't differ more than 40 %. So try to train with resolution 832x832 (batch=64 and ubdivision=16 or 32 or 64) and then try to detect with resolution 1024x1024.

* Also in training dataset your object should be small too.  `train_net_width * train_obj_pixels_width / train_img_width ~= det_net_width * det_obj_pixels_width / det_img_width`

* Remove this line and train again: https://github.com/AlexeyAB/darknet/blob/64aa0180bb74e84a75958b3da0061a9f5615729d/src/data.c#L272

**Comment by shuchitagupta** [comment]
And is it fine to use 1024x1024 resolution for image of size 1920x1080

**Comment by AlexeyAB** [comment]
@shuchitagupta Yes.

----

And if removing this line https://github.com/AlexeyAB/darknet/blob/64aa0180bb74e84a75958b3da0061a9f5615729d/src/data.c#L272 leads to `nan` values. Then change this line to these two:
```
			if (w < 0.01) w = 0.01;
			if (h < 0.01) h = 0.01;
```

**Comment by shuchitagupta** [comment]
@AlexeyAB Thanks

**Comment by shuchitagupta** [comment]
Also, if my width and height is set to 832x832, does that mean my image size used for training should be also 832x832?

**Comment by AlexeyAB** [comment]
@shuchitagupta No. All images will be resized automatically to 832x832 (on both stages: training and detection).
